quinn-proto/src/streams.rs

```diff
@@ -40,21 +41,21 @@ impl Streams {
 
         for dir in Dir::iter() {
             for i in 0..this.max_remote[dir as usize] {
-                this.insert(true, StreamId::new(!side, dir, i));
+                this.insert(None, true, StreamId::new(!side, dir, i));
             }
         }
 
         this
     }
 
-    pub fn open(&mut self, side: Side, dir: Dir) -> Option<StreamId> {
+    pub fn open(&mut self, params: &TransportParameters, side: Side, dir: Dir) -> Option<StreamId> {
```

Comment:
Curious why you decided to put `&TransportParameters` first here as well as in `alloc_remote_stream()` and `insert()`. I'd have put it last as less essential for the purpose of the method. Any thoughts?

Reply:
I default to sorting function parameters by frequency of change descending, but I'm not particularly attached to that pattern. Do you have a strong opinion?

Reply:
I guess not that strong. Not sure what frequency of change means in this context, but doesn't the `params` argument change less here than the other ones?

Reply:
First draft of that comment had an error, meant to type "ascending". The habit arises from my Haskell days, where it plays nicely with currying, but that's not relevant in rust. However, it is consistent with the prior form of the function, since `side` doesn't vary within the lifetime of a connection either.